source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -646,65 +646,43 @@ ShaderNodeImplPtr GlslShaderGenerator::createCompoundImplementation(NodeGraphPtr
 {
-    const string& inputName = input->getName();
-    const string& inputValue = input->getValueString();
-    const string& inputType = input->getType();
+    enumType = nullptr;
 
-    return remapEnumeration(inputName, inputValue, inputType, mappingElement, enumerationType);
-}
-
-ValuePtr GlslShaderGenerator::remapEnumeration(const string& inputName, const string& inputValue, const string& inputType, 
-                                               const InterfaceElement& mappingElement, const TypeDesc*& enumerationType) const
-{
-    enumerationType = nullptr;
-
-    ValueElementPtr valueElem = mappingElement.getChildOfType<ValueElement>(inputName);
-    if (!valueElem)
-    {
-        return nullptr;
-    }
-
-    // Don't convert file names and arrays to integers
-    const TypeDesc* inputTypeDesc = TypeDesc::get(inputType);
-    if (inputTypeDesc->isArray() || inputTypeDesc == Type::FILENAME)
-    {
-        return nullptr;
-    }
-    // Don't convert supported types
-    if (getSyntax().typeSupported(inputTypeDesc))
+    // Early out if not an enum input.
+    const string& enumNames = input.getAttribute(ValueElement::ENUM_ATTRIBUTE);
+    if (enumNames.empty())
     {
         return nullptr;
     }
 
-    // Skip any elements which have no enumerations
-    const string& valueElemEnums = valueElem->getAttribute(ValueElement::ENUM_ATTRIBUTE);
-    if (valueElemEnums.empty())
+    // Don't convert already supported types
+    // or filenames and arrays.
+    const TypeDesc* type = TypeDesc::get(input.getType());
+    if (_syntax->typeSupported(type) ||
+        type == Type::FILENAME || type->isArray())
     {
         return nullptr;
     }
 
-    // Always update the type. For GLSL we always convert to integers,
+    // For GLSL we always convert to integer,
     // with the integer value being an index into the enumeration.
-    enumerationType = TypeDesc::get(TypedValue<int>::TYPE);
+    enumType = Type::INTEGER;
 
     // Update the return value if any was specified. If the value
     // cannot be found always return a default value of 0 to provide some mapping.
-    ValuePtr returnValue = nullptr;
-    if (inputValue.size())
+    int integerValue = 0;
+    if (value.size())
     {
-        int integerValue = 0;
-        StringVec valueElemEnumsVec = splitString(valueElemEnums, ",");
-        auto pos = std::find(valueElemEnumsVec.begin(), valueElemEnumsVec.end(), inputValue);
+        StringVec valueElemEnumsVec = splitString(enumNames, ",");
+        auto pos = std::find(valueElemEnumsVec.begin(), valueElemEnumsVec.end(), value);
         if (pos != valueElemEnumsVec.end())
         {
             integerValue = static_cast<int>(std::distance(valueElemEnumsVec.begin(), pos));
         }
-        returnValue = Value::createValue<int>(integerValue);
     }
-    return returnValue;
+    return Value::createValue<int>(integerValue);
```

Comment:
Doesn't this change the logic so that if the value is empty we return 0 as opposed to not setting a value. 

I think it would end up trying to get a default value and in the end the GLSL syntax would put in a 0 so we get the same result. This code was only left in as a "catch-all" for failure to handle it in other places and I was going to make it throw an exception. 

I does make the logic a bit simpler but it makes it harder to find cases if we're not handling enum strings properly since it won't fall into the catch-all code.